src/cdrizzleapi.c

```diff
@@ -23,425 +23,447 @@
-      PyErr_SetString(PyExc_Exception, driz_error_get_message(&error));
-    return NULL;
-  } else {
-    return Py_BuildValue("i",istat);
-  }
-}
+    const char *kwlist[] = {"source",  "pixmap", "output", "xmin",   "xmax",
+                            "ymin",    "ymax",   "scale",  "kscale", "interp",
+                            "exptime", "misval", "sinscl", NULL};
+
+    /* Arguments in the order they appear */
+    PyObject *oimg, *pixmap, *oout;
+    long xmin = 0;
+    long xmax = 0;
+    long ymin = 0;
+    long ymax = 0;
+    double scale = 1.0;
+    float kscale = 1.0;
+    char *interp_str = "poly5";
+    float ef = 1.0;
+    float misval = 0.0;
+    float sinscl = 1.0;
+
+    PyArrayObject *img = NULL, *out = NULL, *map = NULL;
+    enum e_interp_t interp;
+    int istat = 0;
+    struct driz_error_t error;
+    struct driz_param_t p;
+    integer_t psize[2], osize[2];
+
+    driz_log_handle = driz_log_init(driz_log_handle);
+    driz_log_message("starting tblot");
+    driz_error_init(&error);
+
+    if (!PyArg_ParseTupleAndKeywords(
+            args, keywords, "OOO|lllldfsfff:tblot", (char **)kwlist, &oimg,
+            &pixmap, &oout,                    /* OOO */
+            &xmin, &xmax, &ymin, &ymax,        /* llll */
+            &scale, &kscale, &interp_str, &ef, /* dfsf */
+            &misval, &sinscl)                  /* ff */
+    ) {
+        return NULL;
+    }
 
+    img = (PyArrayObject *)PyArray_ContiguousFromAny(oimg, NPY_FLOAT, 2, 2);
+    if (!img) {
+        driz_error_set_message(&error, "Invalid input array");
+        goto _exit;
+    }
 
-/** --------------------------------------------------------------------------------------------------
+    map = (PyArrayObject *)PyArray_ContiguousFromAny(pixmap, NPY_DOUBLE, 3, 3);
+    if (!map) {
+        driz_error_set_message(&error, "Invalid pixmap array");
+        goto _exit;
+    }
+
+    out = (PyArrayObject *)PyArray_ContiguousFromAny(oout, NPY_FLOAT, 2, 2);
+    if (!out) {
+        driz_error_set_message(&error, "Invalid output array");
+        goto _exit;
+    }
+
+    if (interp_str2enum(interp_str, &interp, &error)) {
+        goto _exit;
+    }
+
+    get_dimensions(map, psize);
+    get_dimensions(out, osize);
+
+    if (psize[0] != osize[0] || psize[1] != osize[1]) {
+        driz_error_set_message(&error,
+                               "Pixel map dimensions != output dimensions");
+        goto _exit;
+    }
+
+    if (xmax == 0) xmax = osize[0];
+    if (ymax == 0) ymax = osize[1];
+
+    driz_param_init(&p);
+
+    p.data = img;
+    p.output_data = out;
+    p.xmin = xmin;
+    p.xmax = xmax;
+    p.ymin = ymin;
+    p.ymax = ymax;
+    p.scale = scale;
+    p.kscale = kscale;
+    p.in_units = unit_cps;
+    p.interpolation = interp;
+    p.ef = ef;
+    p.misval = misval;
+    p.sinscl = sinscl;
+    p.pixmap = map;
+    p.error = &error;
+
+    if (driz_error_check(&error, "xmin must be >= 0", p.xmin >= 0)) goto _exit;
```

Comment:
I think it would be good to always have brackets around conditional code, looping code, etc., as well as that code always being on the next line.  And regardless whether those standards are adopted, I think whatever is adopted should be uniform.  In particular, there are some lines here where the conditional code is on the same line, whereas other lines have the conditional on the next line.

Reply:
I am fine with both approaches. This particular line has a very long condition and I do not like it - I would have used braces and put `goto _exit;` on the next line. However, something like this:
```C
if (x) continue;
```
is fine with me.

**However**, this formatting will not automatically add extra braces for me and the purpose of this style formatting is to automatically format existing code with minimal effort. There will be a lot work done on the `drizzle` code in the future and as we go we can fix many of these things. I did not want to do all of that (including splitting large functions into shorter ones) in this PR.